agent.go

```diff
@@ -233,17 +242,17 @@ func (a *Agent) crankSeparate(shutdown chan struct{}, plugin *runningPlugin) err
 
 		bp.Prefix = plugin.name + "_"
 		bp.Config = plugin.config
-		err := plugin.plugin.Gather(&bp)
-		if err != nil {
-			return err
+		gathererr := plugin.plugin.Gather(&bp)
```

Comment:
Should still use `err` here and in the flush code below. Use this pattern instead:

```go
if err := plugin.plugin.Gather(&bp); err != nil {
  log.Printf("Error in plugin [%s]: %s", plugin.name, err.Error())
}
```